src/Network.ts

```diff
@@ -1,26 +1,131 @@
 import BigNumber from 'bignumber.js';
 
-import { Context } from '~/internal';
+import { Account, Context, transferPolyx, TransferPolyxParams } from '~/internal';
 import { eventByIndexedArgs, eventsByIndexedArgs, transactionByHash } from '~/middleware/queries';
 import { EventIdEnum as EventId, ModuleIdEnum as ModuleId, Query } from '~/middleware/types';
-import { EventIdentifier, ExtrinsicData } from '~/types';
+import {
+  EventIdentifier,
+  ExtrinsicData,
+  NetworkProperties,
+  ProcedureMethod,
+  SubCallback,
+  TxTag,
+  UnsubCallback,
+} from '~/types';
 import { Ensured } from '~/types/utils';
-import { extrinsicIdentifierToTxTag, middlewareEventToEventIdentifier } from '~/utils/conversion';
-import { optionize } from '~/utils/internal';
+import { TREASURY_MODULE_ADDRESS } from '~/utils/constants';
+import {
+  extrinsicIdentifierToTxTag,
+  middlewareEventToEventIdentifier,
+  moduleAddressToString,
+  textToString,
+  u32ToBigNumber,
+} from '~/utils/conversion';
+import { createProcedureMethod, optionize } from '~/utils/internal';
 
 /**
- * Handles all Middleware related functionality
+ * Handles all Network related functionality, including querying for historical events from middleware
  */
-export class Middleware {
+export class Network {
   private context: Context;
 
   /**
    * @hidden
    */
   constructor(context: Context) {
     this.context = context;
+
+    this.transferPolyx = createProcedureMethod(
+      { getProcedureAndArgs: args => [transferPolyx, args] },
+      context
+    );
+  }
+
+  /**
+   * Retrieve the number of the latest block in the chain
+   */
+  public getLatestBlock(): Promise<BigNumber> {
+    return this.context.getLatestBlock();
+  }
+
+  /**
+   * Fetch the current network version (i.e. 3.1.0)
+   */
+  public async getVersion(): Promise<string> {
+    return this.context.getNetworkVersion();
   }
 
+  /**
+   * Retrieve information for the current network
+   */
+  public async getNetworkProperties(): Promise<NetworkProperties> {
+    const {
+      context: {
+        polymeshApi: {
+          runtimeVersion: { specVersion },
+          rpc: {
+            system: { chain },
+          },
+        },
+      },
+    } = this;
+    const name = await chain();
+
+    return {
+      name: textToString(name),
+      version: u32ToBigNumber(specVersion).toNumber(),
+    };
+  }
+
+  /**
+   * Retrieve the protocol fees associated with running a specific transaction
+   *
+   * @param args.tag - transaction tag (i.e. TxTags.asset.CreateAsset or "asset.createAsset")
+   */
+  public getTransactionFees(args: { tag: TxTag }): Promise<BigNumber> {
```

Comment:
I think we should go ahead and change this (and the underlying context method) to `getProtocolFees`. When fetching fees for a `PolymeshTransaction`, they are broken down between `gas` and `protocol`, so we'd be more consistent